pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -735,6 +748,31 @@ func doNetworkInstanceModify(ctx *zedrouterContext,
 	}
 }
 
+func checkNIphysicalPort(ctx *zedrouterContext, status *types.NetworkInstanceStatus) error {
+	// check the NI have the valid physical port binding to
+	label := status.Logicallabel
+	if label != "" && !strings.EqualFold(label, "uplink") &&
+		!strings.EqualFold(label, "freeuplink") {
+		var foundMatch bool
+		pub := ctx.subDeviceNetworkStatus
+		items := pub.GetAll()
+		for _, st := range items {
+			devStatus := st.(types.DeviceNetworkStatus)
+			devPort := getDevPort(&devStatus, label)
```

Comment:
getDevPort --> I know this function exists already - butthis should really be a method of DeviceNetworkStatus

devPort = devStatus.GetPortByIfName(label)

Can you move it there?

That begs the question - if IfName same as logical label? I don't think thats guaranteed.

Reply:
sure. change to the method way.

Reply:
this block we have checked if the label is not null, it's not 'uplink' or 'freeuplink', do we still not guarantee ?

Reply:
@naiming-zededa sc-supermicro-zc2 is running with phylabel != ifname.
AFAIK that means that logicallabel == ifname, but you can log into to it and verify.

Reply:
Yeah you should call
		ifname := types.LogicallabelToIfName(ctx.deviceNetworkStatus,
			label)
then
devPort = ctx.deviceNetworkStatus.GetPortByIfName(ifname)

Reply:
good. ok.